coalib/bears/Bear.py

```diff
@@ -32,17 +51,29 @@ def _is_debugged(bear):
     :return:     True if ``debug_bears`` is ``True`` or if bear name specified
                  in ``debug_bears`` setting match with the bear parameter.
     """
-    if not isinstance(bear, Bear):
-        raise ValueError('Positional argument bear is not an instance of '
-                         'Bear class.')
-    if 'debug_bears' not in bear.section:
-        return False
+    setting = _setting_is_enabled(bear, key='debug_bears')
     try:
-        return bool(bear.section['debug_bears'])
+        value = bool(setting)
     except ValueError:
-        pass
-    return bear.name.lower() in (
-            map(str.lower, bear.section['debug_bears']))
+        value = bear.name.lower() in map(str.lower, setting)
+    return value
+
+
+def _is_profiled(bear):
+    """
+    Check whether the bear is in profile mode according to its section-settings.
+
+    :param bear: Bear object.
+    :return:     current working directory if ``profile`` is ``True``, False
+                 if ``profile`` is ``False`` else return directory path
+                 specified in ``profile``.
+    """
+    setting = _setting_is_enabled(bear, key='profile')
+    try:
+        value = getcwd() if bool(setting) else False
```

Comment:
sheet.  No.  Move the try: except around bool() into _setting_is_enabled, *exactly* as it was within _is_debugged except using `key`.  that part is *common* to both, and puts the bool exception handling in one spot.

Putting `getcwd()` inside the same exception handler for bool(setting) is *wrong*.

It is not more efficient.

It is ugly.

And `getcwd` can throw its own exception, and when someone gets that exception here you are forcing the person debugging this to wonder if it was caused by bool()  (they will probably mentally eliminate that because it will be an OSError), but you are also making it hard for them to use pdb because then cant directly step into getcwd.  They will need to step into bool(setting) first which will take them down the rabbit warren which is StringConverter, which they probably know they dont need to be in.

All because you tried to be 'efficient'.

Save the value from `_setting_is_enabled`, and use `if rv is True:` .  That is the fastest `if` that exists (except 1 == 1), as it is using identity (memory address comparision).  And it is readable.  It tells a story slowly.

Simple code should be simple.

Let the reader keep their brain power unused, so that their brain isnt sore before they reach the really hard code.